Upstream patch dwm-activetagindicatorbar-6.2.diff (https://dwm.suckless.org/patches/activetagindicatorbar/),
applied in this order to dwm 6.5.

diff --git a/dwm.c b/dwm.c
index f1d86b2..d8c0046 100644
--- a/dwm.c
+++ b/dwm.c
@@ -725,9 +725,10 @@ drawbar(Monitor *m)
 		drw_setscheme(drw, scheme[m->tagset[m->seltags] & 1 << i ? SchemeSel : SchemeNorm]);
 		drw_text(drw, x, 0, w, bh, lrpad / 2, tags[i], urg & 1 << i);
 		if (occ & 1 << i)
-			drw_rect(drw, x + boxs, boxs, boxw, boxw,
-				m == selmon && selmon->sel && selmon->sel->tags & 1 << i,
-				urg & 1 << i);
+			drw_rect(drw, x + boxw, bh - 2, w - (2 * boxw + 1), 2,
+			    m == selmon && selmon->sel && selmon->sel->tags & 1 << i,
+			    urg & 1 << i);
+
 		x += w;
 	}
 	w = TEXTW(m->ltsymbol);
